### PR TITLE
Lowlevel pair coalescence quantiles

### PR DESCRIPTION
**Pull request by nspope**
C implementation of `pair_coalescence_quantiles`

### CODE REVIEW

**Comment by nspope** [comment]
This is ready for a look @jeromekelleher

File: c/tskit/trees.c [hunk 1]
```diff
@@ -8516,3 +8539,79 @@ tsk_treeseq_pair_coalescence_counts(const tsk_treeseq_t *self,
...
+int
+tsk_treeseq_pair_coalescence_quantiles(const tsk_treeseq_t *self,
+    tsk_size_t num_sample_sets, const tsk_size_t *sample_set_sizes,
+    const tsk_id_t *sample_sets, tsk_size_t num_set_indexes, const tsk_id_t *set_indexes,
+    tsk_size_t num_windows, const double *windows, tsk_size_t num_bins,
+    const tsk_id_t *node_bin_map, tsk_size_t num_quantiles, double *quantiles,
+    tsk_flags_t options, double *result)
+{
+    int ret = 0;
+    void *params;
```
Comment (jeromekelleher) [inline c/tskit/trees.c:8601]:
```suggestion
    void *params = (void *) quantiles;
```

saves a line (and a bit of human parsing)

**Review by jeromekelleher — APPROVED** [review]
LGTM! Happy to merge when you are.

**Comment by nspope** [comment]
Great, let's get it in!

**Pull request merged**